check_limits.py

```diff
@@ -1,18 +1,14 @@
+def check_in_range(value, min_val, max_val, error_message):
+    if not (min_val <= value <= max_val):
+        print(error_message)
+        return False
+    return True
 
 def battery_is_ok(temperature, soc, charge_rate):
-  if temperature < 0 or temperature > 45:
-    print('Temperature is out of range!')
-    return False
-  elif soc < 20 or soc > 80:
-    print('State of Charge is out of range!')
-    return False
-  elif charge_rate > 0.8:
-    print('Charge rate is out of range!')
-    return False
-
-  return True
-
+    return (check_in_range(temperature, 0, 45, 'Temperature is out of range!') and
+            check_in_range(soc, 20, 80, 'State of Charge is out of range!') and
+            check_in_range(charge_rate, 0, 0.8, 'Charge rate is out of range!'))
```

Comment:
It appears that min value of Charge Rate is assumed to reuse check_in_range function

Reply:
Updated the code : created a new function to check charge rate.